code/modules/projectiles/guns/specialist.dm

```diff
@@ -8,12 +8,161 @@
+	ability_primacy = SPEC_PRIMARY_ACTION_2
+
+/datum/action/item_action/specialist/aimed_shot/New(var/mob/living/user, var/obj/item/holder)
+	..()
+	name = "Aimed Shot"
+	button.name = name
+	button.overlays.Cut()
+	var/image/IMG = image('icons/mob/hud/actions.dmi', button, "sniper_aim")
+	button.overlays += IMG
+	var/obj/item/weapon/gun/rifle/sniper/sniper_rifle = holder_item
+	sniper_rifle.aimed_shot_cooldown = world.time
+
+
+/datum/action/item_action/specialist/aimed_shot/action_activate()
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/H = owner
+	if(H.selected_ability == src)
+		to_chat(H, "You will no longer use [name] with \
+			[H.client && H.client.prefs && H.client.prefs.toggle_prefs & TOGGLE_MIDDLE_MOUSE_CLICK ? "middle-click" : "shift-click"].")
+		button.icon_state = "template"
+		H.selected_ability = null
+	else
+		to_chat(H, "You will now use [name] with \
+			[H.client && H.client.prefs && H.client.prefs.toggle_prefs & TOGGLE_MIDDLE_MOUSE_CLICK ? "middle-click" : "shift-click"].")
+		if(H.selected_ability)
+			H.selected_ability.button.icon_state = "template"
+			H.selected_ability = null
+		button.icon_state = "template_on"
+		H.selected_ability = src
+
+/datum/action/item_action/specialist/aimed_shot/can_use_action()
+	var/mob/living/carbon/human/H = owner
+	if(istype(H) && !H.is_mob_incapacitated() && !H.lying && (holder_item == H.r_hand || holder_item || H.l_hand))
+		return TRUE
+
+/datum/action/item_action/specialist/aimed_shot/proc/use_ability(atom/A)
+	var/mob/living/carbon/human/H = owner
+	if(!istype(A, /mob/living))
+		return
+
+	var/mob/living/M = A
+
+	if(M.stat == DEAD || M == H)
+		return
+
+	var/obj/item/weapon/gun/rifle/sniper/sniper_rifle = holder_item
+	if(world.time < sniper_rifle.aimed_shot_cooldown)
+		return
+
+	if(!check_can_use(M))
+		return
+
+	sniper_rifle.aimed_shot_cooldown = world.time + sniper_rifle.aimed_shot_cooldown_delay
+
+	///Add a decisecond to the default 1.5 seconds for each two tiles to hit.
+	var/distance = round(get_dist(M, H) * 0.5)
+	var/f_aiming_time = sniper_rifle.aiming_time + distance
+	if(HAS_TRAIT(M, TRAIT_SPOTTER_LAZED))
+		f_aiming_time *= 0.5
+
+	var/image/I = image(icon = 'icons/effects/Targeted.dmi', icon_state = "locking-sniper", dir = get_cardinal_dir(M, H))
+	M.overlays += I
+	if(H.client)
+		playsound_client(H.client, 'sound/weapons/TargetOn.ogg', H, 50)
+	playsound(M, 'sound/weapons/TargetOn.ogg', 70, FALSE, 8, falloff = 0.4)
+
+	if(!do_after(H, f_aiming_time, INTERRUPT_ALL|BEHAVIOR_IMMOBILE, NO_BUSY_ICON))
+		M.overlays -= I
+		return
+
+	M.overlays -= I
+
+	if(!check_can_use(M, TRUE))
+		return
+
+	var/obj/item/projectile/P = sniper_rifle.in_chamber
+	P.homing_target = M
+	P.projectile_override_flags |= AMMO_HOMING
+	sniper_rifle.Fire(M, H)
+
+/datum/action/item_action/specialist/aimed_shot/proc/check_can_use(var/mob/M, var/cover_lose_focus)
+	var/mob/living/carbon/human/H = owner
+	var/obj/item/weapon/gun/rifle/sniper/sniper_rifle = holder_item
```

Comment:
please document in the changelog that aimed shot no longer requires being in a prepared position to use.